assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -11815,6 +11844,20 @@ protected int sizeResources(SessionState state) {
                     returnResources.addAll(assignmentService.getAssignmentsForContext((String) state.getAttribute(STATE_CONTEXT_STRING)));
                 }
 
+                //Filter assignments by group
+                if(StringUtils.isNotEmpty(selectedGroup) && !"all".equals(selectedGroup)){
+                    List<Assignment> filteredAssignments = new ArrayList<Assignment>();
+                    for(Assignment assignment : (List<Assignment>) returnResources){
+                        Collection<String> groups = assignment.getGroups();
+                        for(String group : groups){
+                            if(group.endsWith(selectedGroup)){
+                                filteredAssignments.add(assignment);
```

Comment:
lambda might be nice here:
List<Assignment> filteredAssignments = assignments.stream().filter(a -> a.getGroups().stream().anyMatch(g -> g.endsWith("something"))).collect(Collectors.toList());

Reply:
Looks good, thank you!

Reply:
Definitely, I need a fast course of Lambda Expressions, I like them, they are really really powerful.